### PR TITLE
PTNFLY-1952

### PR DESCRIPTION
**Pull request by priley86**
Setup a tool for documenting javascript in PF web components

### CODE REVIEW

**Comment by dtaylor113** [comment]
Can we get a screenshot of the doc pages? :-)

**Comment by priley86** [comment]
Yep! I've got them running here:
https://priley86.github.io/patternfly-webcomponents/

**Comment by dlabrecq** [comment]
1. Is the doc task run separately?
2. I'd like to see the jsDoc located in the dist directory with other generated files, similar to the ngDocs of angular-patternfly.
3. Is the jsDoc something we want to commit to the master branch? I'm thinking this should be generated as part of a Travis build. The docs seem similar to the generated dist files that make PRs difficult to review. For example, we don't commit the ngDocs of angular-patternfly except to the master-dist branch.

FYI, I created PR #12 to introduce a Travis build. I also added code in patternfly-eng-release to build the doc target if the jsdocConfig.json file exists -- available in the next version bump.

File: jsdocConfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "tags": {
+    "allowUnknownTags": true
+  },
+  "opts": {
+    "destination": "./docs"
```
Comment (priley86) [inline jsdocConfig.json]: You can edit docs destination here...

**Comment by priley86** [comment]
To answer the questions...

1. Yes, it can be run with `gulp doc`. We should be able to just add this gulp task to the build pipeline wherever we choose.
2. Currently it writes to the `docs` folder. I've seen this convention used elsewhere as it allows you to quickly serve docs with the Gh-pages Settings. If we have a different approach in Patternfly, i'm fine with using that though. The only problem I see with our Travis build process right now is it doesn't automate releases like the semantic-releases tool does. Automated releases based on commit messages is a really nice feature I'd like to incorporate at some point in the future, but that is a separate issue. To change the destination, just edit the configuration file, `jsdocConfig.json`.
3. If we don't want doc changes showing up in PRs, we can just ignore them in `master` and only include them in `dist`. I'd just like to ensure we can serve them to Github pages automatically though (if you have another approach).

Let me know any changes you'll need here...

**Comment by dlabrecq** [comment]
I'm fairly certain it doesn't matter to GitHub where the docs folder is located? 

We're already pushing from master-dist to GitHub pages, so there's no compelling reason to commit generated docs to master. I'll let you make the necessary changes here.

Don't want to rat hole into another topic, but we can easily create a release of patternfly-webcomponents using the same scripts from the patternfly-eng-release repo. From what I read about the semantic-releases tool, it appears to perform similar tasks to our release.sh and publish-ghpages.sh scripts, except we have more control over the process. For example, we perform far more testing.

If we're going to add another repo to the release process, I would strongly urge that we have a similar approach for all. (Adding patternfly-webcomponents would mean having 5 repos to release and possibly 8 total after we split up patternfly.) Having a single patternfly-eng-release repo helps a lot with the release process. Otherwise, it's too much without a dedicated release engineer.

File: jsdocConfig.json (hunk 1 above)
Comment (dlabrecq) [inline jsdocConfig.json]: I'd like to see doc under the dist folder, similar to the ngDocs of angular-patternfly.

Comment (dlabrecq) [inline jsdocConfig.json]: I don't believe we should commit the JS doc folder to master, but allow Travis to build it via master-dist. The Travis build is already setup to build the jsDoc via PR #12.

Comment (priley86) [inline jsdocConfig.json]: ok so just to clarify, move `docs` to `dist/docs`? anything else? Should we merge your Travis PR first? I'm a little unclear on next steps.

Comment (priley86) [inline jsdocConfig.json]: I'm fine with using the same patterns as the other repos. This process works fine...

Comment (dlabrecq) [inline jsdocConfig.json]: I don't know that it matters if PR #12 is merged, first. The master-dist branch will be created either way. If docs is under dist/docs, they will automatically be added to the commit and merged to master-dist.

Comment (dlabrecq) [inline jsdocConfig.json]: But, feel free to merge PR #12 so it will be ready.

Comment (priley86) [inline jsdocConfig.json]: Merged #12 and updated docs path to `dist/docs`

**Comment by dlabrecq** [comment]
@dtaylor113 , do you want to review? I'd like to merge to test this with the Travis build.

**Comment by dlabrecq** [comment]
Spoke with Dave and he's ok with the PR. Merging...

**Comment by priley86** [comment]
thanks @dlabrecq

**Pull request merged**